tokio-timer/src/deadline.rs

```diff
@@ -0,0 +1,159 @@
+//! Docs
+
+use Sleep;
+
+use futures::{Future, Poll, Async};
+
+use std::time::Instant;
+
+/// Allows a given `Future` up to the specified deadline.
+#[must_use = "futures do nothing unless polled"]
+#[derive(Debug)]
+pub struct Deadline<T> {
+    future: T,
+    sleep: Sleep,
+}
+
+/// Error returned by `Deadline` future.
+#[derive(Debug)]
+pub struct DeadlineError<T> {
+    kind: Kind<T>,
+}
+
+/// Deadline error variants
+#[derive(Debug)]
+enum Kind<T> {
+    /// Inner future returned an error
+    Inner(T),
+
+    /// The deadline elapsed.
+    Elapsed,
+
+    /// Timer returned an error.
+    Timer(::Error),
+}
+
+impl<T> Deadline<T> {
+    pub fn new(future: T, deadline: Instant) -> Deadline<T> {
+        Deadline::new_with_sleep(future, Sleep::new(deadline))
+    }
+
+    pub(crate) fn new_with_sleep(future: T, sleep: Sleep) -> Deadline<T> {
+        Deadline {
+            future,
+            sleep,
+        }
+    }
+
+    /// Gets a reference to the underlying future in this deadline.
+    ///
+    /// # Return
+    ///
+    /// The function returns `None` if the inner future has already been
```

Comment:
This comment seems incorrect. The return value is always a reference to the inner `Future`.